platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java

```diff
@@ -428,7 +428,11 @@ public boolean onScale(ScaleGestureDetector detector) {
         transform.zoomBy(detector.getScaleFactor(), focalPoint.x, focalPoint.y);
       } else if (quickZoom) {
         // around center map
-        transform.zoomBy(detector.getScaleFactor(), uiSettings.getWidth() / 2, uiSettings.getHeight() / 2);
+        float scaleFactor = detector.getScaleFactor();
+        if (scaleFactor > 0.65 && scaleFactor < 1.35) {
```

Comment:
1. yes, will be smoother, I initially left this out because I was seeing negative values and thought that clamping those would give a reversed zoom, but after investigating these occurrences it's always followed with values resembling a reverse zoom so clamping here is valid.

```
01-03 18:13:12.049 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: Before 0.70612484
01-03 18:13:12.050 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: After 0.70612484
01-03 18:13:12.066 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: Before 0.994616
01-03 18:13:12.067 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: After 0.994616
01-03 18:13:12.081 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: Before -0.1821847
01-03 18:13:12.082 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: After 0.65
01-03 18:13:12.100 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: Before 0.623573
01-03 18:13:12.101 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: After 0.65
01-03 18:13:12.120 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: Before 0.7911937
01-03 18:13:12.121 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: After 0.7911937
01-03 18:13:12.134 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: Before 0.88355064
01-03 18:13:12.135 18481-18481/com.mapbox.mapboxsdk.testapp E/MapGestureDetector$ScaleGestureListener: After 0.88355064
```

2. purely based on values I'm seeing locally here on 3 devices. I have now moved them to the MapboxConstants file. FWIW the 0.65 value is the maximum scale we can zoom out, the 1.35 resembls the maximum scale we can zoom in. Feel free to test out changing these values while testing out this PR.